src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -551,17 +551,13 @@ private static Details getDetails(DetailAST ast) {
             switch (ast.getType()) {
                 case TokenTypes.LITERAL_TRY:
                 case TokenTypes.LITERAL_CATCH:
-                case TokenTypes.LITERAL_FINALLY:
```

Comment:
This can be removed because the code for `getDetailsForTryCatchFinally` gives the same results as `getDetailsForOthers` for the `LITERAL_FINALLY` token. This is because `finally` must always have `{}`s and the constant tree structure.

````
        |      |  `--LITERAL_FINALLY -> finally [5:2]
        |      |      `--SLIST -> { [5:10]
...
        |      |          |--SEMI -> ; [6:9]
        |      |          `--RCURLY -> } [7:0]
````

````
        private static Details getDetailsForTryCatchFinally(DetailAST ast) {
            final DetailAST lcurly;
            DetailAST nextToken;
            final int tokenType = ast.getType();
            if (false) {
...
            }
            else {
                nextToken = ast.getNextSibling(); // == null
                lcurly = ast.getLastChild(); // == `{` (always)

            }

            final boolean shouldCheckLastRcurly;
            if (true) {
                shouldCheckLastRcurly = true;
                nextToken = getNextToken(ast); // is always what comes after `}`
            }
            else {
...
            }

            final DetailAST rcurly = lcurly.getLastChild(); // == `}`
            return new Details(lcurly, rcurly, nextToken, shouldCheckLastRcurly);
        }
````

````
        private static Details getDetailsForOthers(DetailAST ast) {
            DetailAST rcurly = null;
            final DetailAST lcurly;
            final int tokenType = ast.getType();
            if (false) {
...
            }
            else {
                lcurly = ast.findFirstToken(TokenTypes.SLIST); // == `{` (always)
                if (true) {
                    // SLIST could be absent if method is abstract
                    rcurly = lcurly.getLastChild(); // == `}` (always)
                }
            }
            return new Details(lcurly, rcurly, getNextToken(ast), true);
        }
````

In short, 1 is:
````
lcurly = ast.getLastChild();
shouldCheckLastRcurly = true;
nextToken = getNextToken(ast);
rcurly = lcurly.getLastChild();
````

2 is:
````
lcurly = ast.findFirstToken(TokenTypes.SLIST);
rcurly = lcurly.getLastChild();
nextToken = getNextToken(ast)
````

The only difference is lcurly (`ast.getLastChild()` versus `ast.findFirstToken(TokenTypes.SLIST)` which is equivalent by the tree.